docs/compatibility.md

```diff
@@ -0,0 +1,185 @@
+  + Forward compatibility: Window has closed, no need for downgrade anymore.
+  + Backward compatibility: Maintain upgrade until compatibility window closes.
+
+### Addition Change
+_Characteristics of a addition change:_
+- A new feature is introduced, with new semantics.
+- No forward compatibility guarantees when feature is used.
+- Warn on forward incompatibility if serializing new feature during the forward compatibility window.
+
+_Example:_ Change enum `Enum` which has value `<A>` to add a new value `<B>`.
+
+_Compatibility Protocol:_
+
+![Addition Change Protocol Diagram](images/addition_change.png)
+
+_Description:_
+- 1/1/23: Create bytecode_v1 from libStablehlo@v1
+- 1/7/2023: Add enum value `<B>` which has new semantics in libStablehlo@v2
+  + No need to create a new version of the op or to write upgrade / downgrade hooks.
+  + Backward compatibility: libStablehlo@v2 can handle `<A>` from bytecode_v1.
+  + Forward compatibility: libStablehlo@v1 can handle `<A>` from bytecode_v2, but `<B>` from bytecode_v2 doesn't fall under provided compatibility guarantees because it's a new feature. During the forward compatibility window, `--stablehlo-compatibility-check` will emit a warning if it sees `<B>`.
+- 1/29/23: Remove warning in libStablehlo@v3
+  + Backward compatibility: Nothing changes from 1/7/2023.
+  + Forward compatibility: Window has closed, no need to emit warning anymore.
+
+### Deprecation Change
+_Characteristics of a deprecation change:_
+- Remove a feature with backwards compatibility
+- Prevent creation of new bytecode using feature
+- Clean up after backward compatibility window
+
+_Example:_ Change enum `Enum` which has values `<A>` and `<B>` to remove value `<B>`.
+
+_Compatibility Protocol:_
+
+![Deprecation Change Protocol Diagram](images/deprecation_change.png)
+
+_Description:_
+- 1/1/23: Create bytecode_v1 from libStablehlo@v1
+
+- 1/7/23: Add compatibility check to prevent serialization of `<B>` in libStablehlo@v2
+  + Backward compatibility: libStablehlo@v2 can deserialize `<A>` and `<B>` from bytecode_v1.
+  + Forward compatibility: libStablehlo@v1 can handle `<A>` from bytecode_v2. libStablehlo@v2 prevents the serialization of `<B>`.
+- 6/8/23: Remove `<B>` deserialization and enum value in libStablehlo@v3.
+  + Ensure that numeric value of other enums does not change due to deletion.
+  + Backward compatibility: Window has closed, no need to maintain `<B>`.
+  + Forward compatibility: No changes required, libStablehlo@v2 can handle all libStablehlo@v3 programs.
+
+### Versioned Op Change
+_Characteristics of a change that requires versioning:_
+- A change is introduced that cannot be upgraded / downgraded
+- Create new version of legacy op
+- A combination of Addition/Deprecation Protocols from there
+
+_Example:_ Changing the signature and semantics to an op, adding an operand.
+
+_Compatibility Protocol:_
+
+![Versioned Op Change Protocol Diagram](images/versioned_op_change.png)
```

Comment:
On the diagram, you use "bc" to refer to "bytecode", right? Let's expand the abbreviation.